tests/telemetry/test_telemetry.py

```diff
@@ -160,3 +161,104 @@ def test_logs_after_fork(run_python_code_in_subprocess):
 
     assert status == 0, err
     assert err == b""
+
+
+def test_app_started_error_handled_exception(test_agent_session, run_python_code_in_subprocess):
+    code = """
+import logging
+logging.basicConfig()
+
+from ddtrace import tracer
+from ddtrace.filters import TraceFilter
+
+class FailingFilture(TraceFilter):
+    def process_trace(self, trace):
+       raise Exception("Exception raised in trace filter")
+
+tracer.configure(
+    settings={
+        "FILTERS": [FailingFilture()],
+    }
+)
+
+# generate and encode span
+tracer.trace("hello").finish()
+"""
+    _, stderr, status, _ = run_python_code_in_subprocess(code)
+    assert status == 0, stderr
+    assert b"Exception raised in trace filter" in stderr
+
+    events = test_agent_session.get_events()
+
+    assert len(events) == 3
+
+    # Same runtime id is used
+    assert events[0]["runtime_id"] == events[1]["runtime_id"]
+    assert events[0]["request_type"] == "app-closing"
+    assert events[1]["request_type"] == "app-dependencies-loaded"
+    assert events[2]["request_type"] == "app-started"
+    assert events[2]["payload"]["error"]["code"] == 1
+    assert (
+        "ddtrace/internal/processor/trace.py/trace.py:221: error applying processor FailingFilture()"
+        in events[2]["payload"]["error"]["message"]
+    )
+
+
+def test_app_started_error_unhandled_exception(test_agent_session, run_python_code_in_subprocess):
+    env = os.environ.copy()
+    env["DD_SPAN_SAMPLING_RULES"] = "invalid_rules"
+    env["DD_INSTRUMENTATION_TELEMETRY_ENABLED"] = "true"
+
+    _, stderr, status, _ = run_python_code_in_subprocess("import ddtrace", env=env)
+    assert status == 1, stderr
+    assert b"Unable to parse DD_SPAN_SAMPLING_RULES=" in stderr
+
+    events = test_agent_session.get_events()
+
+    assert len(events) == 2
+
+    # Same runtime id is used
+    assert events[0]["runtime_id"] == events[1]["runtime_id"]
+    assert events[0]["request_type"] == "app-closing"
+    assert events[1]["request_type"] == "app-started"
+    assert events[1]["payload"]["error"]["code"] == 1
+    assert (
+        "/test.py:1: Unable to parse DD_SPAN_SAMPLING_RULES='invalid_rules'" in events[1]["payload"]["error"]["message"]
+    )
+
+
+def test_integration_error(test_agent_session, run_python_code_in_subprocess):
+    code = """
+import logging
+logging.basicConfig()
+
+import sqlite3
+# patch() of the sqlite integration assumes this attribute is there
+# removing it should cause patching to fail.
+del sqlite3.connect
+
+from ddtrace import patch, tracer
+patch(raise_errors=False, sqlite3=True)
+tracer.trace("test").finish()
```

Comment:
Do we need 242 and 243?

Reply:
Tested, we do... in order to flush the events?  The writer's `on_shutdown()` doesn't send them?